tms/file_manager/file_manager.py

```diff
@@ -0,0 +1,209 @@
+            try:
+                return fpath.stat().st_mtime <= threshold_time
+            except FileNotFoundError:
+                return False
+
+        # Plain file (or symlink to a file)
+        else:
+            try:
+                return fpath.stat().st_mtime <= threshold_time
+            except FileNotFoundError:
+                return False
+
+    async def act(self, fpath: Path) -> None:
+        """Perform action on filepath, if the file is old enough."""
+        if not fpath.exists():
+            raise FileNotFoundError(fpath)
+
+        if self.precheck is not None:
+            try:
+                if not await self.precheck(fpath):
+                    LOGGER.warning(
+                        f"precheck failed for {fpath=} -- will try again later in {ENV.TMS_FILE_MANAGER_INTERVAL}"
+                    )
+                    return
+            except Exception:
+                LOGGER.exception(f"precheck raised for {fpath}")
+                return
+
+        if not self.is_old_enough(fpath):
+            LOGGER.info(
+                f"no action -- filepath not older than {self.age_threshold} seconds {fpath=}"
+            )
+            return
+
+        LOGGER.info(f"performing action {self.action} on {fpath}")
+        try:
+            self.action(fpath)
+        except Exception:
+            LOGGER.exception(f"action failed for {fpath}")
+
+
+# -----------------------------------------------------------------------------
+# MAIN_LIST
+# -----------------------------------------------------------------------------
+
+
+MAIN_LIST: list[FileManager] = [
+    #
+    # ex: 2025-8-26.tms.jel
+    FileManager(
+        str(JELFileLogic.parent / f"*{JELFileLogic.suffix}"),
+        action=action_rm,
+        age_threshold=ENV.JOB_EVENT_LOG_MODIFICATION_EXPIRY,
+        precheck=JELFileLogic.is_no_longer_used,
+    ),
+    #
+    # ex: ewms-taskforce-TF-685e6219-e85461b3-f8dc0d3c-6e4a5d72
+    FileManager(
+        str(TaskforceDirLogic.parent / f"{TaskforceDirLogic.prefix}*"),
+        action=partial(action_tar_gz, dest=ENV.JOB_EVENT_LOG_DIR),
+        age_threshold=ENV.TASKFORCE_DIRS_EXPIRY,
+    ),
+    #
+    # ex: ewms-taskforce-TF-685e6219-e85461b3-f8dc0d3c-6e4a5d72.tar.gz
+    FileManager(
+        str(TaskforceDirLogic.parent / f"{TaskforceDirLogic.prefix}*.tar.gz"),
+        action=action_rm,
+        age_threshold=ENV.TASKFORCE_DIRS_TAR_EXPIRY,
+    ),
+]
+
+
+# -----------------------------------------------------------------------------
+# Runner
+# -----------------------------------------------------------------------------
+
+
+async def run() -> None:
+    """Run the file manager loop."""
+    await asyncio.sleep(60)
+
+    while True:
+
+        LOGGER.info("inspecting filepaths...")
+
+        for fm in MAIN_LIST:
+            LOGGER.info(f"searching filepath pattern: {fm.fpattern}")
+            for fpath in glob.glob(fm.fpattern):
```

Comment:
It might be useful to throw a try/except block around this for loop, to make sure no exceptions escape to bring down the file manager.